modules/gateway/nodes.go

```diff
@@ -102,68 +102,126 @@ func (g *Gateway) requestNodes(conn modules.PeerConn) error {
 	return nil
 }
 
-// threadedNodeManager tries to keep the Gateway's node list healthy. As long
-// as the Gateway has fewer than minNodeListSize nodes, it asks a random peer
-// for more nodes. It also continually pings nodes in order to establish their
-// connectivity. Unresponsive nodes are aggressively removed.
-func (g *Gateway) threadedNodeManager() {
-	if g.threads.Add() != nil {
-		return
-	}
-	defer g.threads.Done()
+// permanentNodePurger is a thread that runs throughout the lifetime of the
+// gateway, purging unconnectable nodes from the node list in a sustainable
+// way.
+func (g *Gateway) permanentNodePurger(closeChan chan struct{}) {
+	defer close(closeChan)
 
 	for {
+		// Start by sleeping for 10 minutes. This means that no nodes will be
+		// purged for the first 10 minutes that Sia is running, and that any
+		// failures to get nodes will be met by 10 minutes of waiting.
+		//
+		// At most one node will be contacted every 10 minutes. This minimizes
+		// the total amount of keepalive traffic on the network.
 		select {
-		case <-time.After(5 * time.Second):
+		case <-time.After(10 * time.Minute):
 		case <-g.threads.StopChan():
+			// The gateway is shutting down, close out the thread.
 			return
 		}
 
+		// Get a random node for scanning.
 		g.mu.RLock()
 		numNodes := len(g.nodes)
-		peer, err := g.randomPeer()
+		node, err := g.randomNode()
 		g.mu.RUnlock()
 		if err != nil {
-			// can't do much until we have peers
+			// Most likely the error indicates that there are no nodes in the
+			// node list. Wait until the next iteration to try again.
```

Comment:
It would be helpful to assert this assumption by logging the error if it is not `errNoPeers`.